File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/overload/drive.java
package org.firstinspires.ftc.teamcode.overload;
import com.acmerobotics.dashboard.FtcDashboard;
import com.acmerobotics.dashboard.telemetry.MultipleTelemetry;
import com.acmerobotics.roadrunner.Pose2d;
import com.acmerobotics.roadrunner.PoseVelocity2d;
import com.acmerobotics.roadrunner.Vector2d;
import com.qualcomm.hardware.modernrobotics.ModernRoboticsI2cColorSensor;
import com.qualcomm.robotcore.eventloop.opmode.OpMode;
import com.qualcomm.robotcore.eventloop.opmode.TeleOp;
import com.qualcomm.robotcore.eventloop.opmode.LinearOpMode;
// Drake was here :)
import com.qualcomm.robotcore.hardware.DcMotor;

import org.firstinspires.ftc.robotcore.external.Telemetry;
import org.firstinspires.ftc.teamcode.MecanumDrive;


//import org.firstinspires.ftc.teamcode.code.util.BackupHardware;


@TeleOp(name="Drive", group = "TeleOp")
public class drive extends LinearOpMode {

    //Class def




    @Override
    public void runOpMode() throws InterruptedException {
        while (opModeInInit()){}

        while (opModeIsActive()) {
            MecanumDrive mecDrive = new MecanumDrive(hardwareMap, new Pose2d(0, 0, Math.toRadians(0)));
            double deflator;
            deflator = gamepad1.left_bumper && gamepad1.right_bumper ? 0.5 : gamepad1.left_bumper ? 0.7 : 1;


            mecDrive.setDrivePowers(new PoseVelocity2d(
                        new Vector2d(
                                -gamepad1.left_stick_y*deflator,
                                -gamepad1.left_stick_x*deflator
                        ),
                        -gamepad1.right_stick_x*deflator
                    ));

        }
    }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/Kotlin_Bromine_Arya/Opmodes/Testing$Tuning/Auto/Localizer/drive.kt
package org.firstinspires.ftc.teamcode.Kotlin_Bromine_Arya.Opmodes.`Testing$Tuning`.Auto.Localizer

import com.acmerobotics.dashboard.FtcDashboard
import com.acmerobotics.dashboard.telemetry.MultipleTelemetry
import com.acmerobotics.roadrunner.Pose2d
import com.qualcomm.robotcore.eventloop.opmode.LinearOpMode
import com.qualcomm.robotcore.eventloop.opmode.TeleOp
import org.firstinspires.ftc.teamcode.Kotlin_Bromine_Arya.Auto.Localizer.Localizer
import org.firstinspires.ftc.teamcode.Kotlin_Bromine_Arya.Tele.SubSystems.Drive

@TeleOp(name = "RRLocalizer", group = "Linear OpMode")
class drive : LinearOpMode(){
    override fun runOpMode() {
        telemetry = MultipleTelemetry(telemetry, FtcDashboard.getInstance().telemetry)

        val drive = Localizer(hardwareMap, Pose2d(0.0, 0.0, 0.0))
        val move = Drive(hardwareMap, drive)
        Drive.driveType = Drive.DriveType.Manual

        waitForStart()

        while (opModeIsActive() && !isStopRequested) {
            drive.update()

            move.drive(arrayOf( gamepad1.left_stick_x, -gamepad1.left_stick_y, gamepad1.right_stick_x))

            telemetry.addData("x",drive.xPos)
            telemetry.addData("y",drive.yPos)
            telemetry.addData("h",Math.toDegrees(drive.heading.toDouble()))
            telemetry.update()
        }
    }
}